server/app/org/scalaexercises/exercises/controllers/AuthenticationModule.scala

```diff
@@ -45,8 +44,8 @@ trait AuthenticationModule { self: ProdInterpreters ⇒
   def AuthenticatedUser(block: User ⇒ Future[Result])(implicit userOps: UserOps[ExercisesApp], transactor: Transactor[Task]) =
     Secure(AuthenticationAction.async { request ⇒
       userOps.getUserByLogin(request.userId).runFuture flatMap {
-        case Xor.Right(Some(user)) ⇒ block(user)
-        case _                     ⇒ Future.successful(BadRequest("User login not found"))
+        case Right(Some(user)) ⇒ block(user)
```

Comment:
We should rename all occurrences of `block` to `thunk` or something more descriptive as block can be misinterpreted as this having blocking semantics in the context of `scala.concurrent.Future`